Add Maximum Backups for Amazon S3

aaronhuisinga opened the pull request:
Add a new configuration option for users to be able to choose the
maximum amount of backups to store in their Amazon S3 bucket. After the
maximum amount of backups are stored, the system will properly delete
the amount necessary to trim to the maximum, deleting oldest to newest.

aaronhuisinga committed:
Add Maximum Backups for Amazon S3

Add a new configuration option for users to be able to choose the
maximum amount of backups to store in their Amazon S3 bucket. After the
maximum amount of backups are stored, the system will properly delete
the amount necessary to trim to the maximum, deleting oldest to newest.

aaronhuisinga commented:
Just a note: I didn't write up any extra tests for this functionality. Everything is running just fine on several of my test environments, but I'd obviously recommend writing tests to be 100% sure before merging.

schickling commented:
Hey Aaron,

I'd really love to merge this feature but please also provide some testcases for this!

aaronhuisinga commented:
To be honest I'm terrible with phpunit, but I'll try and get some written up over the weekend.
I'm also working on adding maximum local backups as well as gzip support. Those will be different pull requests.

schickling commented:
Awesome. It's definitely worth it investing some time in learning test driven development! :+1: 

I'm really looking forward to merge the other pull requests!

aaronhuisinga committed:
Add Compression & Remove Local File After Sync

Use gzip to compress the database backups before uploading them to
Amazon S3. This should help to save some considerable space. Also
remove local file once it has been synced to Amazon S3 to save server
space.

aaronhuisinga committed:
Cherry Pick Upstream Changes

Several package changes have been made since v0.5.0. Merge in cherry
picked changes so the package can stay as close as possible.

aaronhuisinga committed:
Finish Upstream Merge

Merge final cherry picks from v0.5.0

aaronhuisinga committed:
Merge in support for Laravel 5

Config references and config file publication needed to be updated.

aaronhuisinga committed:
Update Readme

aaronhuisinga committed:
Update composer dependencies for Laravel 5

aaronhuisinga committed:
Update AWS dependency for Laravel 5

aaronhuisinga committed:
Give AWS SDK hard version

aaronhuisinga committed:
Revert "Give AWS SDK hard version"

This reverts commit 703203df955adf8cd6555b375f9fc4d1fd942aa0.

aaronhuisinga committed:
Rename Package to Prevent Versioning Conflicts
